### PR TITLE
Ordering of property declaration attributes (ObjC)

### PR DESCRIPTION
**Pull request by Krizai**
Based on existing pull request ( https://github.com/uncrustify/uncrustify/pull/217 )
I've extracted the topic related stuff from it and added support for a few new attributes.

### CODE REVIEW

**Comment by mihaipopescu** [comment]
Nicely done! Still, in order to prove your changes are working the way you want they have to be covered by tests. Can you add some ?

**Comment by gmaurel** [comment]
We really need somebody who handled the very numerous issues with ObjC.
Could you have a look on these?

**Comment by Krizai** [comment]
A few fixes + test added

**Comment by Krizai** [comment]
As for the issues - I don't promise anything, but will try to check.

**Comment by Krizai** [comment]
Is it possible to merge this or do you want any more changes ?

File: src/combine.cpp [hunk 1]
```diff
@@ -5517,6 +5514,213 @@ static void handle_oc_message_send(chunk_t *os)
...
+        chunk_t *next = chunk_get_next(os);
+        chunk_t *open_paren = NULL;
+        chunk_t *close_paren = NULL;
+        
+        std::vector<ChunkGroup> thread_chunks;      // atomic/nonatomic
+        std::vector<ChunkGroup> readwrite_chunks;   // readwrite, readonly
+        std::vector<ChunkGroup> ref_chunks;         // retain, copy, assign, weak, strong, unsafe_unretained
+        std::vector<ChunkGroup> getter_chunks;      // getter
+        std::vector<ChunkGroup> setter_chunks;      // setter
+        std::vector<ChunkGroup> nullability_chunks; // nonnull/nullable
```
Comment (daniel-beard) [inline src/combine.cpp]: Missing `null_resettable`

File: src/combine.cpp [hunk 2]
```diff
@@ -5517,6 +5514,213 @@ static void handle_oc_message_send(chunk_t *os)
...
+{
+    if (cpd.settings[UO_mod_sort_oc_properties].b)
+    {
+        typedef std::vector<chunk_t*> ChunkGroup;
+
+        chunk_t *next = chunk_get_next(os);
+        chunk_t *open_paren = NULL;
+        chunk_t *close_paren = NULL;
+        
+        std::vector<ChunkGroup> thread_chunks;      // atomic/nonatomic
```
Comment (daniel-beard) [inline src/combine.cpp]: This group is also missing `class` property support

**Comment by Krizai** [comment]
@daniel-beard Thank you! Will add.

**Comment by cheeseonhead** [comment]
When used together with `sp_after_comma` there will be two spaces after the commas. I'm guessing you're adding the commas there by yourself, and I don't think that's required.

**Comment by daniel-beard** [comment]
@Krizai `null_unspecified` is also a valid nullable annotation.

**Comment by Krizai** [comment]
@daniel-beard @cheeseonhead  Please check the update

**Pull request merged**